File: routes/eligibilidad.js
import express from "express";
import pool from "../db.js";

const router = express.Router();

router.get("/:usuarioId", async (req, res) => {
  const { usuarioId } = req.params;
  try {
    const result = await pool.query(
      "SELECT elegible FROM usuarios WHERE id = $1",
      [usuarioId]
    );

    if (result.rows.length === 0) {
      return res.status(404).json({ mensaje: "Usuario no encontrado" });
    }

    res.json({ elegible: result.rows[0].elegible });
  } catch (error) {
    console.error(error);
    res.status(500).json({ mensaje: "Error al verificar elegibilidad" });
  }
});

export default router;

File: routes/eligibilidad.py
from fastapi import APIRouter, HTTPException, Depends
from sqlalchemy.orm import Session
from services.api.database import get_db
from services.api.models import User
import logging

router = APIRouter()
logger = logging.getLogger(__name__)

@router.get("/usuarios/{usuario_id}")
def verificar_elegibilidad(usuario_id: int, db: Session = Depends(get_db)):
    """Verifica si un usuario es elegible"""
    try:
        user = db.query(User).filter(User.id == usuario_id).first()
        
        if not user:
            raise HTTPException(status_code=404, detail="Usuario no encontrado")
        
        # Aquí puedes agregar lógica de elegibilidad
        return {
            "usuario_id": usuario_id,
            "elegible": user.is_verified,
            "email": user.email
        }
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"Error verificando elegibilidad: {str(e)}")
        raise HTTPException(status_code=500, detail=f"Error al verificar elegibilidad: {str(e)}")
